### PR TITLE
Bugfix: Re-execute dagstermill Op

### PR DESCRIPTION
**Pull request by motuzov**
## Summary & Motivation
An exception raises than Re-executes a dagstermill Op.
Failure condition: Attempted to access KnownExecutionState but it was not provided at context creation.
[Issue 13013](https://github.com/dagster-io/dagster/issues/13013)

## How I Tested These Changes
1. Create a simple tow stage pipeline.
2. Run whole pipeline from Launchpad.
3. Re-execute the second stage.
 
fix-Re-execute

### CODE REVIEW

**Comment by smackesey** [comment]
Thanks @motuzov, I'm running our CI on this and if it passes then I think we're good to merge.

File: python_modules/libraries/dagstermill/dagstermill/manager.py [hunk 1]
```diff
@@ -190,6 +197,7 @@ def reconstitute_job_context(
                     job_context.for_step(
                         cast(ExecutionStep, execution_plan.get_step_by_key(step_key))
                     ),
+                    known_state=known_state
```
Comment (smackesey) [inline python_modules/libraries/dagstermill/dagstermill/manager.py:200]: `known_state` needs to be passed to `job_context.for_step` (it's currently passed to `cast`)

File: python_modules/libraries/dagstermill/dagstermill/manager.py [hunk 2]
```diff
@@ -175,6 +176,12 @@ def reconstitute_job_context(
             # Set this flag even though we're not in test for clearer error reporting
             raise_on_error=True,
         ) as job_context:
+            known_state = None
+            if dagster_run.parent_run_id:
+                known_state = KnownExecutionState.build_for_reexecution(
+                    instance=instance,
+                    parent_run=instance.get_run_by_id(dagster_run.parent_run_id)
```
Comment (smackesey) [inline python_modules/libraries/dagstermill/dagstermill/manager.py:183]: `instance.get_run_by_id(dagster_run.parent_run_id)` needs to become `check.not_none(instance.get_run_by_id(dagster_run.parent_run_id))`

**Review by smackesey — CHANGES_REQUESTED** [review]
Sorry it took me so long to report back-- CI surfaced some errors (see inlines). Also, I recommend making sure `make black`, `make ruff`, and `make pyright` pass before pushing another commit (these will catch most CI failures).

File: python_modules/libraries/dagstermill/dagstermill/manager.py (hunk 2 above)
Comment (motuzov) [reply python_modules/libraries/dagstermill/dagstermill/manager.py:183]: Thanks @smackesey, I'll fix it soon.

Comment (smackesey) [reply python_modules/libraries/dagstermill/dagstermill/manager.py:183]: I actually have made the changes locally and can push to your branch if given permission (these last linting steps can be a real hassle to go back and forth on)-- would you consider doing this: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests

**Comment by smackesey** [comment]
Thanks @motuzov, I was able to directly commit to the branch to fix remaining lint errors and am now merging!

**Review by smackesey — APPROVED** [review]

**Pull request merged**